src/main/java/seedu/mindmymoney/command/ListCommand.java

```diff
@@ -67,25 +69,71 @@ private boolean hasIncomeListFlag() {
      */
-    public String expenditureListToString() {
+    public String expenditureListToString() throws MindMyMoneyException {
         int indexOfList = 1;
         String listInString = "";
+        if (listInput.equals(FLAG_OF_EXPENSES)) {
+            listInString = printListString(indexOfList, listInString);
+        } else {
+            String[] inputArray = GeneralFunctions.parseInput(listInput);
+            if (!inputArray[1].equals("")) {
+                if (!isValidInput(inputArray[1])) {
+                    throw new MindMyMoneyException("Date has to be in \"dd/mm/yyyy\", \"mm/yyyy\" or \"yyyy\" format!");
+                }
+                for (Expenditure i : expenditureList.expenditureListArray) {
+                    if (i.getTime().contains(inputArray[1])) {
+                        listInString += indexOfList + ". $" + i.getAmount() + " was spent on " + i.getDescription()
+                                + "(" + i.getCategory() + ") " + "using " + i.getPaymentMethod()
+                                + " [" + i.getTime() + "]" + "\n";
+                        indexOfList++;
+                    }
+                }
+            } else {
+                listInString = printListString(indexOfList, listInString);
+            }
+        }
+        assert listInString.length() != 0 : "Return string should be non-empty";
+        return listInString;
+    }
 
+    /**
+     * Formats the output of expenses in list.
+     *
+     * @param index To obtain the numbering when listing the expenses.
+     * @param listInString String where the content of output is appended to.
+     * @return
+     */
+    public String printListString(int index, String listInString) {
         for (Expenditure i : expenditureList.expenditureListArray) {
-            listInString += indexOfList + ". $" + i.getAmount() + " on " + i.getDescription() + ". Paid using "
-                    + i.getPaymentMethod() + " [" + i.getCategory() + "]" + " [" + i.getTime() + "]\n";
-            indexOfList++;
+            listInString += index + ". $" + i.getAmount() + " was spent on " + i.getDescription() + "("
+                    + i.getCategory() + ") " + "using " + i.getPaymentMethod() + " [" + i.getTime() + "]" + "\n";
+            index++;
         }
-        assert listInString.length() != 0 : "Return string should be non-empty";
         return listInString;
     }
 
+    /**
+     * Checks if date input format is valid.
+     *
+     * @param input The string of the date input.
+     * @return true if format is valid, false otherwise.
+     */
+    public static boolean isValidInput(String input) {
+        if (input.matches(ValidationRegexTypes.VALIDATION_REGEX_D)
+                || input.matches(ValidationRegexTypes.VALIDATION_REGEX_M)
+                || input.matches(ValidationRegexTypes.VALIDATION_REGEX_Y)) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
Minor thing, but can just return false
```suggestion
        return false;
```

Reply:
resolved